contents/testing/security-testing.md

```diff
@@ -0,0 +1,146 @@
+
+- __Vulnerability Assessment__: The system is scanned and analyzed for security vulnerabilities.
+- __Penetration Testing__: The system undergoes analysis and attack from simulated malicious attackers.
+- __Runtime Testing__: The system undergoes analysis and security testing from an end-user.
+- __Code Review__: The system code undergoes a detailed review and analysis looking specifically for security vulnerabilities.
+
+<box type="info">Note that risk assessment, which is commonly listed as part of security testing, is not included in this list. That is because a risk assessment is not actually a test but rather the analysis of the perceived severity of different risks (software security, personnel security, hardware security, etc.) and any mitigation steps for those risks.</box>
+
+## Why Have Security Testing?
+
+Security testing ensures that a system is secure. Since security requirements differ depending on the goals of the project, it is often hard to define the concept of what it means to be secure. One definition is as follows[^3]: <box type="definition">Security means that authorized access is granted to protected data and unauthorized access is restricted.</box> This definition encompasses two major aspects; first is the protection of data and the second is access to that data. Whether the system is desktop or web-based, we will loosely define security as revolving around the two aforementioned aspects.
+
+Due to the logical limitations of security testing however, passing security testing is not an indication that no flaws exist or that the system adequately satisfies the security requirements[^4]. Why have security testing then? In brief, a system would be more secure with security testing than without.
+
+<box type="warning">There is more to the concept of security than the working definition used here.</box>
+
+## Security Testing in Action
+
+The consensus is that it is much more costly to include security testing after software implementation or deployment[^5]. It is necessary then to involve security testing in all phases of a Software Development Life Cycle [(SDLC)](https://se-education.org/se-book/processModels/). A conceptual example of such an involvement is as follows (taken from [Guru99.com](https://www.guru99.com/what-is-security-testing.html#3)):
+
+<center>
+<pic src="security-testing/Sec-Test-Fig-1.png" width="800" alt="Example">
+Figure 1 - Example of Security Testing during a SDLC.
+</pic>
+</center>
+
+<br>
+
+To understand this in more detail, we can deconstruct the application of security testing into its manifestation in different stages of software development.
+
+### At A Glance
+
+<center>
+<pic src="security-testing/Sec-Test-Fig-2.png" width="650" alt="Workflow">
+Figure 2 - Testing Framework Workflow from the OWASP Testing Guide.
+</pic>
+</center>
+
+<br>
+
+The following series of steps adapted from the OWASP Testing Guide[^1] provide general applications of security testing in a generic SDLC process.
```

Comment:
Consider having this paragraph before the image. It may be a bit abrupt to lead with an image.